### ISSUE TITLE
Images Aren't Running through Fixed Position Flux Check Properly

### ISSUE THREAD

**Issue by ycendes** [transientskp/tkp#463]
@dariocarbone6 has run into this issue as well, so hopefully he can add whatever else he can to this issue.

Basically, I ran into an issue recently when taking images from MSSS (ie LOFAR survey) and wanting to find the flux at a given position, ie

> pyse.py --fixed-posns=[[250.423,36.461]] M250_33_mosaic-band7_sub.fits_miriad_convol_circbeam_BPA.fits

If you run the above, you get the following error:

```
Processing M250_33_mosaic-band7_sub.fits_miriad_convol_circbeam_BPA.fits_with_metadata.fits (file 1 of 1).
WARNING:tkp.accessors.fitsimage:End time not specified or unreadable
ERROR:tkp.accessors.fitsimage:Frequency not specified in FITS
Traceback (most recent call last):
  File "/home/ycendes/aartfaacenv/bin/pyse.py", line 10, in <module>
    execfile(__file__)
  File "/home/ycendes/tkp/tkp/bin/pyse.py", line 358, in <module>
    print run_sourcefinder(files, options),
  File "/home/ycendes/tkp/tkp/bin/pyse.py", line 295, in run_sourcefinder
    ff = open_accessor(filename, beam=beam, plane=0)
  File "/home/ycendes/tkp/tkp/accessors/__init__.py", line 71, in open
    return Accessor(path, *args, **kwargs)
  File "/home/ycendes/tkp/tkp/accessors/requiredatts.py", line 61, in __call__
    obj = type.__call__(cls, *args, **kwargs)
  File "/home/ycendes/tkp/tkp/accessors/fitsimage.py", line 32, in __init__
    self.freq_eff, self.freq_bw = self.parse_frequency()
  File "/home/ycendes/tkp/tkp/accessors/fitsimage.py", line 145, in parse_frequency
    raise TypeError(msg)
TypeError: Frequency not specified in FITS
```

Now the image in question (which is just in my home directory) did not have this information in its header originally, as often happens for MSSS images, but it was a simple enough thing to add this info in using Dario's pyse_on_aartfaac.py script, which is as follows:

```
#import numpy
import optparse
import os
import pyfits

usage = "USAGE: python pyse_on_aartfaac.py input_file"
parser = optparse.OptionParser()
(opts, args) = parser.parse_args()
input_files = []
for i in range (0, len(args)):
	input_files.append(args[i])

for line in input_files:
#	nam = line.split('.')[0]+'.'+line.split('.')[1]
#	name = nam+'.fits'
#	os.system('image2fits in=' + line + ' out=' + name)

	hdulist = pyfits.open(line) # open a FITS file
	hdr = hdulist[0].header # the primary HDU header
	hdr.update('RESTFRQ', 1.57E+08, 'Rest Frequency (Hz)')
#	hdr.update('BPA', 0., 'Beam position angle')
#	hdr.update('BMAJ', 1., 'Beam major axis')
#	hdr.update('BMIN', 1., 'Beam minor axis')
	pyfits.writeto(line+'_with_metadata.fits', pyfits.getdata(line), hdr)

#	os.system('pyse.py --detection=5 --analysis=3 --regions --residuals --deblend --deblend-thresholds=300 --grid=100 --csv '+nam+'_with_metadata.fits')
```
So basically even though the correct information is in the header, the sourcefinder isn't finding it for some reason.

**Comment by gijzelaerr** [comment]
I suggest have a look at the code where the exception is raised: https://github.com/transientskp/tkp/blob/master/tkp/accessors/fitsimage.py#L144

There is no clear standard (or not that I know) for putting the frequency in the header, so we tried to make a parser for often encountered formats. You guys set `RESTFRQ`, but if you look at the code you can see that is only used if `hdr['ctype3'] not in ('FREQ', 'VOPT')` or `hdr['ctype4'] in ('FREQ', 'VOPT')`. So **if** one of these conditions is true different fields are parsed.

I'm not sure what file you are trying to work with, so can't have a look.

**Comment by ycendes** [comment]
Ah, sorry!  The image is in my home directory, M250_33_mosaic-band7_sub.fits_miriad_convol_circbeam_BPA.fits (and, similarly, M250_33_mosaic-band7_sub.fits_miriad_convol_circbeam_BPA.fits_with_metadata.fits ).  I don't know if @dariocarbone6 has some other examples of this or not.
